ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -168,6 +170,14 @@ private Block createBlock(
 
       throwIfStopped();
 
+      Long baseFee = null;
```

Comment:
is it not missing the "final"? and wouldn't it be better to use an optional instead of a null?

Reply:
Cannot be final since i update the value.
It removes the need to have if - else block there.
The base fee field in the block header is null for non EIP-1559 blocks. The `getBaseFee() `method wraps it in an `Optional`